### DISCUSSION TITLE
Add more feature flags to the Rust crate

### DISCUSSION THREAD

**Discussion by keremnymn** [lance-format/lance#2627] [Ideas]
Hi,

I see that the [Rust crate](https://docs.rs/lancedb/latest/lancedb/) has a lot of dependencies. For example if I want to use local connection, why am I installing all the AWS dependencies? I think more feature flags can be added for separate connection methods to reduce the dependency overhead.

**Comment by wjones127** [comment]
We can eliminate a few dependencies, but many of them are required for the breadth of functionality we offer. I created an issue to track adding some feature flags. If you have additional dependencies you think we might be able to eliminate, feel free to comment there:

https://github.com/lancedb/lance/issues/2646
